src/controllers/keyboard/keyboardcontrollerpreset.cpp

```diff
@@ -0,0 +1,242 @@
+                continue;
+            }
+
+            // Check whether we need no modifier or shift modifier for the scancode lookup
+            QStringList modifiers = layoutUtils::getModifiersFromKeysequence(keyseqs[0]);
+            bool onlyShift = modifiers.size() == 1 && modifiers[0] == "Shift";
+            Qt::KeyboardModifier modifier = onlyShift ? Qt::ShiftModifier : Qt::NoModifier;
+
+            // Retrieve scancodes (usually just one, sometimes two,
+            // due to multiple keys sharing the same character)
+            QList<int> scancodes = layoutUtils::findScancodesForCharacter(originalKeyseqLayout, key.at(0), modifier);
+
+            // Warn user if any scancodes where found for this character
+            if (scancodes.isEmpty()) {
+                qWarning() << "Couldn't find any scancodes for character"
+                           << key.at(0) << "on" << keyseqsRawI->byPositionOf
+                           << (onlyShift ? "with" : "without") << "shift."
+                           << "Skipping mapping for" << configKey;
+                continue;
+            }
+
+            // Remove original key sequence
+            keyseqs.clear();
+
+            // Iterate through all found scancodes
+            for (const int& scancode : scancodes) {
+
+                // Get KbdKeyChar
+                const KbdKeyChar *keyChar = layoutUtils::getKbdKeyChar(layout, (unsigned char) scancode, modifier);
+                QChar character(keyChar->character);
+
+                // If key is not dead, reconstruct key sequence with translated character. Otherwise,
+                // warn user about key that won't work.
+                if (!keyChar->isDead) {
+                    QString modifiersString = modifiers.join("+");
+                    if (!modifiersString.isEmpty()) {
+                        modifiersString += "+";
+                    }
+                    keyseqs.append(modifiersString + character);
+                } else {
+                    qWarning() << "Can't use key with scancode " << scancode
+                               << " because it's a dead key on layout '" << layoutName
+                               << "'. Please specify <keyseq lang=\"" << layoutName
+                               << "\" scancode=\"" << scancode << "\"> in the keyboard preset file (group: "
+                               << configKey.group << ").";
+                    keyseqs[0] = "";
+                }
+            }
+        }
+
+        for (QString& keyseq : keyseqs) {
+            QString key = layoutUtils::getCharFromKeysequence(keyseq);
+            QStringList mods = layoutUtils::getModifiersFromKeysequence(keyseq);
+
+            // Clear keyseq
+            keyseq = "";
+
+            // NOTE: This order must be the same as in
+            // ...   KeyboardEventFilter::getKeySeq()
+            if (mods.contains("Shift")) keyseq += "Shift+";
```

Comment:
You use "+" for concatenating key codes at various places in your code. It would be a good idea to use a constant for this purpose instead of repeating the literal "+".